example/gmres/ex_real_A.cpp

```diff
@@ -0,0 +1,129 @@
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Jennifer Loe (jloe@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#include<math.h>
+#include"KokkosKernels_IOUtils.hpp"
+#include<Kokkos_Core.hpp>
+#include<Kokkos_Random.hpp>
+#include<KokkosBlas.hpp>
+#include<KokkosBlas3_trsm.hpp>
+#include<KokkosSparse_spmv.hpp>
+
+#include"gmres.hpp"
+
+int main(int argc, char *argv[]) {
+
+  typedef double                            ST;
+  typedef int                               OT;
+  typedef Kokkos::DefaultExecutionSpace     EXSP;
+
+  using ViewVectorType = Kokkos::View<ST*,Kokkos::LayoutLeft, EXSP>;
+
+  std::string filename("bcsstk09.mtx"); // example matrix
+  std::string ortho("CGS2"); //orthog type
+  int m = 50; //Max subspace size before restarting.
+  double convTol = 1e-10; //Relative residual convergence tolerance.
+  int cycLim = 50; //Maximum number of times to restart the solver. 
+
+  for (int i=1;i<argc;++i) {
+    const std::string& token = argv[i];
+    if (token == std::string("--filename")) filename = argv[++i];
+    if (token == std::string("--max-subsp")) m = std::atoi(argv[++i]);
+    if (token == std::string("--max-restarts")) cycLim = std::atoi(argv[++i]);
+    if (token == std::string("--tol")) convTol = std::stod(argv[++i]);
+    if (token == std::string("--ortho")) ortho = argv[++i];
+    if (token == std::string("--help") || token == std::string("-h")){
+      std::cout << "Kokkos GMRES solver options:" << std::endl
+        << "--filename    :  The name of a matrix market (.mtx) file for matrix A (Default bcsstk09.mtx)." << std::endl
+        << "--max-subsp   :  The maximum size of the Kyrlov subspace before restarting (Default 50)." << std::endl
+        << "--max-restarts:  Maximum number of GMRES restarts (Default 50)." << std::endl
+        << "--tol         :  Convergence tolerance.  (Default 1e-10)." << std::endl
+        << "--ortho       :  Type of orthogonalization. Use 'CGS2' or 'MGS'. (Default 'CGS2')" << std::endl
+        << "--help  -h    :  Display this help message." << std::endl 
+        << "Example Call  :  ./Gmres.exe --filename Laplace3D100.mtx --tol 1e-5 --max-subsp 100 " << std::endl << std::endl;
+      return 0; }
+  }
+  std::cout << "File to process is: " << filename << std::endl;
+  std::cout << "Convergence tolerance is: " << convTol << std::endl;
+
+  //Initialize Kokkos AFTER parsing parameters:
+  Kokkos::initialize();
+  {
+
+  // Read in a matrix Market file and use it to test the Kokkos Operator.
+  KokkosSparse::CrsMatrix<ST, OT, EXSP> A = 
+    KokkosKernels::Impl::read_kokkos_crst_matrix<KokkosSparse::CrsMatrix<ST, OT, EXSP>>(filename.c_str()); 
+
+  int n = A.numRows();
+  ViewVectorType X("X",n); //Solution and initial guess
+  ViewVectorType Wj("Wj",n); //For checking residuals at end.
+  ViewVectorType B(Kokkos::ViewAllocateWithoutInitializing("B"),n);//right-hand side vec
```

Comment:
`Kokkos::ViewAllocateWithoutInitializing` has been deprecated last week, it would be good to replace it with `Kokkos::alloc(Kokkos::WithoutInitializing, label)` see [here](https://github.com/kokkos/kokkos/pull/4155)